packages/docs-reanimated/docs/reanimated-babel-plugin/about.md

```diff
@@ -0,0 +1,239 @@
+  console.log('Hello from ArrowFunctionExpression');
+};
+```
+
+- Object Methods
+
+```ts
+const obj = {
+  foo() {
+    'worklet';
+    console.log('Hello from ObjectMethod');
+  },
+};
+```
+
+## Autoworkletization
+
+To reduce boilerplate code and provide a safer API, Reanimated Babel Plugin detects automatically whether a function should be workletized. Thanks to that, you don't need to add the `'worklet'` directive to your callbacks:
+
+```ts
+const style = useAnimatedStyle(() => {
+  // You don't need to add the 'worklet' directive here,
+  // since plugin detects this callback as autoworkletizable.
+  return {
+    width: 100,
+  };
+});
+```
+
+This isn't limited to `useAnimatedStyle` hook - Reanimated Babel Plugin autoworkletizes all callbacks for the API of Reanimated. The whole list can be found in the [plugin source code](https://github.com/software-mansion/react-native-reanimated/blob/main/plugin/src/autoworkletization.ts).
+
+Keep in mind that in more advanced use cases, you might still need to manually mark a function as a worklet.
+
+### Referencing worklets
+
+You can define worklets **before** they are used and the plugin will autoworkletize them too:
+
+```ts
+function foo() {
+  // You don't need to add
+  // the 'worklet' directive here.
+  return { width: 100 };
+}
+
+// You don't need to define an inline function here,
+// a reference is enough.
+const style = useAnimatedStyle(foo);
+```
+
+### Objects aggregating worklets
+
+In some APIs, like `useAnimatedScrollHandler` you can pass an object that contains worklets instead of a function:
+
+```ts
+const handlerObject = {
+  // You don't need to mark these methods as worklets.
+  onBeginDrag() {
+    console.log('Dragging...');
+  },
+  onScroll() {
+    console.log('Scrolling...');
+  },
+};
+
+const handler = useAnimatedScrollHandler(handlerObject);
+```
+
+### Workletizing whole files (experimental)
+
+You can mark a file as a workletizable file by adding the `'worklet'` directive to the top of the file. This will make all _top-level_ entities workletized. This can come in handy for files that contain multiple worklets.
+
+```ts
+'worklet';
+
+function foo() {
+  // Function 'foo' will be autoworkletized.
+  return { width: 100 };
+}
+
+function bar() {
+  // Function 'bar' will be autoworkletized.
+  function foobar() {
+    // Function 'foobar' won't since it's not defined in top-level scope.
+    console.log("I'm not a worklet");
+  }
+  return { width: 100 };
+}
+```
+
+## Limits of autoworkletization
+
+The plugin cannot infer whether a function is autoworkletizable or not in some contexts.
+
+### Imports
+
+When importing a function from another file or a module and using it as a worklet, you must manually add the `'worklet'` directive to the function:
+
+```ts
+// foo.js
+export function foo() {
+  'worklet'; // Won't work without it.
+  return {
+    width: 100,
+  };
+}
+
+// bar.ts
+import { foo } from './foo';
+const style = useAnimatedStyle(foo);
+```
```

Comment:
I'd swap places of these two file snippets. I think it is more clear if we firstly define the context (importing function to use it in useAnimatedStyle) and then show a snippet (similar to many others before), but with a forced 'worklet' directive